### ISSUE TITLE
Watch issues with node_modules.

### ISSUE THREAD

**Issue by CrossR** [onivim/oni#2217]
Currently, Oni on Linux seems to be watching many many more files than it needs to be.

Quoting @dkns from Discord:

>it seems like somewhere along the way we run watch command on whole oni repo? which including node_modules is over 100k files. Excluding node_modules it's 1158 files and on ubuntu 16.04 default max_user-watches is 8192

I'm not sure where to look into that, but I can't see any reason we'd need to watch `node_modules`.
I think this is only relevant to running a development build.

**Comment by badosu** [comment]
@CrossR I agree some filtering should happen, but we should have a more general method than only not watching `node_modules`.

Perhaps not watching VCS ignored files like `.gitignore`?

**Comment by Akin909** [comment]
@CrossR @badosu if this relates to the file system watcher then I think this might be a duplicate of #2149, which led to turning the fsWatcher of by default so I think depending on @dkns version of oni this should be not quite resolved till the issue is fully resolved but the fsWatcher should no longer be on

**Comment by bryphe** [comment]
Thanks for moving this to an issue, @CrossR !

Yes, sounds like either `typescript` or `webpack` is listening to the `node_modules` folder from the `yarn start` command. There's a few different things that happen with `yarn start`:
```
        "start": "concurrently --kill-others \"yarn run start-hot\" \"yarn run watch:browser\" \"yarn run watch:plugins\"",
```

- `yarn run start-hot` - starts electron with `ONI_WEBPACK_LOAD` environment variable
- `watch:browser` - uses `webpack-dev-server` against our `webpack-development.config.js`
- `watch:plugins` - watches all our other plugin projects.

I would guess it's either `watch:browser` or `watch:plugins` - would be helpful to isolate which one is causing the problem.

A workaround is to build w/o using the live-reload:
`yarn build-debug`
`yarn launch`

That doesn't use any watchers, so shouldn't hit this issue - just a little bit more of a pain than the `yarn start` strategy, though, since you have to run these commands after every code change.

**Comment by badosu** [comment]
So this is an issue with the development watcher then

**Comment by dkns** [comment]
I was looking at this yesterday and it still happens when you set max user watches to ~15000 but at 25000 it's fine. So it's not watching whole node_modules, just some subset of it.
